src/indexer/handleEvent.ts

```diff
@@ -363,23 +368,45 @@ async function handleEvent(
 
       for (let i = startIndex; i < startIndex + bitmap.itemsPerRow; i++) {
         const status = bitmap.getStatus(i);
-        const statusString = ApplicationStatus[status];
+        const statusString = ApplicationStatus[status] as Application["status"];
         const application = await db
           .collection<Application>(`rounds/${event.address}/applications`)
-          .updateById(i.toString(), (application) => ({
-            ...application,
-            status: statusString as Application["status"],
-            statusUpdatedAtBlock: event.blockNumber,
-          }));
+          .updateById(i.toString(), (application) => {
+            const newApplication = {...application}
```

Comment:
To help the type checker catch issues before the `return` statement, please declare the type for `newApplication`, as in:

```ts
const newApplication: Application =  {....application}
```